app/src/main/java/org/jellyfin/mobile/webapp/ConnectionHelper.kt

```diff
@@ -0,0 +1,208 @@
+package org.jellyfin.mobile.webapp
+
+import android.view.KeyEvent
+import android.view.View
+import android.view.inputmethod.EditorInfo
+import android.view.inputmethod.InputMethodManager
+import android.webkit.WebView
+import android.widget.Button
+import android.widget.EditText
+import android.widget.FrameLayout
+import android.widget.TextView
+import androidx.annotation.StringRes
+import androidx.core.content.getSystemService
+import androidx.core.view.doOnNextLayout
+import androidx.core.view.isVisible
+import androidx.core.view.postDelayed
+import androidx.lifecycle.lifecycleScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import okhttp3.HttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import org.jellyfin.mobile.AppPreferences
+import org.jellyfin.mobile.R
+import org.jellyfin.mobile.WebappActivity
+import org.jellyfin.mobile.databinding.ConnectServerBinding
+import org.jellyfin.mobile.utils.Constants
+import org.jellyfin.mobile.utils.Constants.SERVER_INFO_PATH
+import org.jellyfin.mobile.utils.requestNoBatteryOptimizations
+import org.json.JSONException
+import org.json.JSONObject
+import timber.log.Timber
+import java.io.IOException
+
+class ConnectionHelper(private val activity: WebappActivity) {
+    private val appPreferences: AppPreferences get() = activity.appPreferences
+    private val rootView: FrameLayout get() = activity.rootView
+    private val webView: WebView get() = activity.webView
+
+    private var cachedInstanceUrl: HttpUrl? = null
+    var connected = false
+        private set
+
+    private val connectServerBinding: ConnectServerBinding by lazy { ConnectServerBinding.inflate(activity.layoutInflater, rootView, false) }
```

Comment:
```suggestion
    private val connectServerBinding: ConnectServerBinding by lazy {
         ConnectServerBinding.inflate(activity.layoutInflater, rootView, false)
     }
```
(indentation might be bad)/ Line is a bit long so I'd split it